src/Evolution/EventsAndDenseTriggers/DenseTriggers/Or.hpp

```diff
@@ -0,0 +1,136 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <algorithm>
+#include <limits>
+#include <memory>
+#include <pup.h>
+#include <pup_stl.h>
+#include <utility>
+#include <vector>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "Evolution/EventsAndDenseTriggers/DenseTrigger.hpp"
+#include "Options/Options.hpp"
+#include "Options/ParseOptions.hpp"
+#include "Parallel/CharmPupable.hpp"
+#include "Time/EvolutionOrdering.hpp"
+#include "Time/TimeStepId.hpp"
+#include "Utilities/Algorithm.hpp"
+#include "Utilities/Registration.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+namespace Tags {
+struct TimeStepId;
+}  // namespace Tags
+/// \endcond
+
+namespace DenseTriggers {
+template <typename TriggerRegistrars>
+class Or;
+
+namespace Registrars {
+using Or = Registration::Registrar<DenseTriggers::Or>;
+}  // namespace Registrars
+
+/// \ingroup EventsAndTriggersGroup
+/// Trigger when any of a collection of triggers triggers.
+template <typename TriggerRegistrars>
+class Or : public DenseTrigger<TriggerRegistrars> {
+ public:
+  /// \cond
+  Or() = default;
+  explicit Or(CkMigrateMessage* /*unused*/) noexcept {}
+  using PUP::able::register_constructor;
+  WRAPPED_PUPable_decl_template(Or);  // NOLINT
+  /// \endcond
+
+  using Result = typename DenseTrigger<TriggerRegistrars>::Result;
+
+  static constexpr Options::String help =
+      "Trigger when any of a collection of triggers triggers.";
+
+  explicit Or(std::vector<std::unique_ptr<DenseTrigger<TriggerRegistrars>>>
+                  triggers) noexcept
+      : triggers_(std::move(triggers)) {}
+
+  using is_triggered_argument_tags =
+      tmpl::list<Tags::TimeStepId, Tags::DataBox>;
+
+  template <typename DbTags>
+  Result is_triggered(const TimeStepId& time_step_id,
+                      const db::DataBox<DbTags>& box) const noexcept {
+    const evolution_less<double> before{time_step_id.time_runs_forward()};
+    Result result{false, time_step_id.time_runs_forward()
+                             ? std::numeric_limits<double>::infinity()
+                             : -std::numeric_limits<double>::infinity()};
+    for (const auto& trigger : triggers_) {
```

Comment:
have I understood correctly that the reason the for loop doesn't exit after the first `is_triggered` is to set `next_check` correctly? Up to you if you think it's worth adding a comment to make sure someone doesn't add a `break` in the future :)